pkg/graveler/errors.go

```diff
@@ -23,7 +23,7 @@ var (
 	ErrNoMergeBase                  = errors.New("no merge base")
 	ErrInvalidRef                   = fmt.Errorf("ref: %w", ErrInvalidValue)
 	ErrInvalidCommitID              = fmt.Errorf("commit id: %w", ErrInvalidValue)
-	ErrInvalidBranchID              = fmt.Errorf("branch id: %w", ErrInvalidValue)
+	ErrInvalidBranchID              = fmt.Errorf("branch id: %w can contain only letters, numbers, underscores, and dashes (not at the beginning)", ErrInvalidValue)
```

Comment:
This will have the form

> branch id: invalid value: can contain only letters, numbers, underscores, and dashes (not at the beginning)

which looks somewhat different from `ErrInvalidRef` and friends.  Consider

```suggestion
	ErrInvalidBranchID              = fmt.Errorf("branch id must consist of letters, digits, underscores and dashes, and cannot start with a dash: %w", ErrInvalidValue)
```

However I would also like to question whether we need this validation almost anywhere except during creation.  Specifically, if I try to **create** a branch named '-not=valid--` then it should fail.  But anywhere else I only _fetch_ a branch, and I think I should be just as happy to be told "branch not found".